<a href="https://colab.research.google.com/github/GadAsa1/Nautica-Home-Project/blob/main/Nautica_Home_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Nautica Home Data Analysis

# Loading the libaries

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
! git clone https://github.com/GadAsa1/Nautica-Home-Project.git # Connect to Github

Cloning into 'Nautica-Home-Project'...
remote: Enumerating objects: 25, done.
remote: Counting objects: 100% (25/25), done.
remote: Compressing objects: 100% (25/25), done.
remote: Total 25 (delta 7), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (25/25), 1.23 MiB | 6.23 MiB/s, done.


# First Table - 'Items'

## 1. Loading the Table

In [3]:
df_items = pd.read_excel("/content/Nautica-Home-Project/Nautica Home - Items Catalog.xlsx")

In [4]:
print(f"This Dataframe has {df_items.shape[0]} rows over {df_items.shape[1]} columns")

This Dataframe has 1579 rows over 104 columns


In [5]:
df_items.head()

מק"ט          ברקוד                תאור  \
0        231200200      231200200  צלחת עיקרית 27 ס"מ   
1  231200200-02-27  7297619005943  צלחת עיקרית 27 ס"מ   
2  231200200-19-27  7297619005936  צלחת עיקרית 27 ס"מ   
3        231200201      231200201  צלחת ראשונה 23 ס"מ   
4  231200201-02-20  7297619005967  צלחת ראשונה 23 ס"מ   

                           תאור לועזי סטטוס טיפוס מנוהל מלאי? שינוי תאור?  \
0  10.5' Shallow Plate - DINNER PLATE  פעיל     R           Y         NaN   
1  10.5' Shallow Plate - DINNER PLATE  פעיל     R           Y         NaN   
2  10.5' Shallow Plate - DINNER PLATE  פעיל     R           Y         NaN   
3      8' Shallow Plate - SALAD PLATE  פעיל     R           Y         NaN   
4      8' Shallow Plate - SALAD PLATE  פעיל     R           Y         NaN   

   מנוהל מס. סידוריים?  רכש אינטרנט?  ... מק"ט חסוי? דרופשיפינג  \
0                  NaN           NaN  ...        NaN        NaN   
1                  NaN           NaN  ...        NaN        NaN   
2                  NaN           NaN  ...        NaN        NaN   
3                  NaN           NaN  ...        NaN        NaN   
4                  NaN           NaN  ...        NaN        NaN   

  חיוב נלווה להשכרות מידות טקסטיל מטבע קנייה 2  מחיר קנייה 2  שם ספק מועדף 2  \
0                NaN          NaN          NaN             0             NaN   
1                NaN          NaN          NaN             0             NaN   
2                NaN          NaN          NaN             0             NaN   
3                NaN          NaN          NaN             0             NaN   
4                NaN          NaN          NaN             0             NaN   

  ספק מועדף 2 מטבע של סה"כ מ.קנייה  סה"כ מחיר קנייה  
0         NaN                  NaN                0  
1         NaN                  NaN                0  
2         NaN                  NaN                0  
3         NaN                  NaN                0  
4         NaN                  NaN                0  

[5 rows x 104 columns]

## 2. Cleanup the Table

### 2.1 Empty columns and rows

In [6]:
df_items.isnull().sum()

מק"ט                       0
ברקוד                     35
תאור                       0
תאור לועזי               262
סטטוס                      0
                        ... 
מחיר קנייה 2               0
שם ספק מועדף 2          1579
ספק מועדף 2             1579
מטבע של סה"כ מ.קנייה    1579
סה"כ מחיר קנייה            0
Length: 104, dtype: int64

Note : The dataset contain NaN values so cleanup in this area is requierd.☝ 

In [7]:
#pd.options.display.max_rows = 999

In [8]:
df_items = df_items.dropna(axis = 1, how = 'all') # drop an axis completely empty

In [9]:
df_items =df_items.dropna(axis = 0, how = 'all')  # drop an row completely empty

 Cleaning the empty columns and rows ☝

In [10]:
df_items.shape

(1579, 73)

This Dataframe has now 1579 rows over 73 columns, so we deleted 31 columns and 0 rows.

### 2.2 Columns with single value

In [11]:
single_value_col = df_items.columns[df_items.nunique() <=1]
single_value_col

Index(['סטטוס', 'טיפוס', 'מנוהל מלאי?', 'שינוי תאור?', 'יח' קניה/מכירה',
       'יח' מפעל', 'שעור המרה', 'פריט מרכז?', 'מטבע המחירון', 'מחיר סיטונאות',
       'ת.  תמחיר', 'מטבע מחיר מינימום', 'מחיר מינימום', 'FOB',
       'סכום מירבי לתיקון', 'שנה', 'מדיניות מחירים', 'מאפיין סופר פארם',
       'סדר ליקוט', 'חודשי DC מותרים', 'חודשי DC לאחסון', 'מינימום ימי תוקף',
       'מחיר קנייה 2'],
      dtype='object')

In [12]:
df_items = df_items.drop(single_value_col, axis = 1)  # Cleaning the columns with a simgle value

In [13]:
df_items.shape

(1579, 50)

This Dataframe has now 1579 rows over 50 columns, so we deleted another 23 columns.

### 2.3 Duplicate Rows

In [14]:
duplicate_rows = df_items[df_items.duplicated()].shape[0]
print(f"This Dataframe has {duplicate_rows} duplicate rows ")

This Dataframe has 0 duplicate rows 


### 2.4 Unnecessary columns and rows

In [15]:
filter = df_items.isnull().sum().sort_values(ascending=False)
filter[ filter > 0 ]

שם סדרה               1509
פרמטר 16 למוצר        1509
שם סרגל מידות         1491
סרגל מידות            1491
מידות טקסטיל          1424
קניין                 1419
פרמטר 9 למוצר         1324
הרכב בד/חומר          1246
GSM                   1227
תת משפחה              1152
קטגוריית על           1140
תאור חדר              1134
צבע טקסטיל            1123
ספק מועדף             1081
שם ספק                1081
עונה                  1077
מטבע                  1074
תאריך מחיר אחרון      1074
מקור מחיר אחרון       1074
פרמטר‎‎‎ ‎4            995
כמות באריזה            954
דגם                    931
תאריך יציאה אחרון      905
תאריך כניסה אחרון      710
תמונה                  695
תאור לועזי             262
תאור פריט מרכז         221
פריט מרכז              221
מטבע קניה              198
צבעים                  186
מידות                  181
ברקוד                   35
תנועה אחרונה במפעל      11
dtype: int64

In [16]:
filter_1 = (df_items.isnull().sum()/df_items.shape[0]).sort_values(ascending=False)
filter_1[ filter_1 > 0  ] # percentage of NaNs value

שם סדרה               0.955668
פרמטר 16 למוצר        0.955668
שם סרגל מידות         0.944269
סרגל מידות            0.944269
מידות טקסטיל          0.901837
קניין                 0.898670
פרמטר 9 למוצר         0.838505
הרכב בד/חומר          0.789107
GSM                   0.777074
תת משפחה              0.729576
קטגוריית על           0.721976
תאור חדר              0.718176
צבע טקסטיל            0.711210
ספק מועדף             0.684611
שם ספק                0.684611
עונה                  0.682077
מטבע                  0.680177
תאריך מחיר אחרון      0.680177
מקור מחיר אחרון       0.680177
פרמטר‎‎‎ ‎4           0.630146
כמות באריזה           0.604180
דגם                   0.589614
תאריך יציאה אחרון     0.573148
תאריך כניסה אחרון     0.449652
תמונה                 0.440152
תאור לועזי            0.165928
תאור פריט מרכז        0.139962
פריט מרכז             0.139962
מטבע קניה             0.125396
צבעים                 0.117796
מידות                 0.114630
ברקוד                 0.022166
תנועה אח

In order to continue the process of cleaning the data, we met with the CTO and presented to him:


*   Columns with missing values.
*   Columns thaht had similar meaning.
*   Columns that seemed irrelevant to the purpose of the project.

After this meeting we decided to:
*   Remove the rows where `Central_Item` is equal to `Y`.
*   Remove another 28 columns.

So we get the following data 👇

In [22]:
df_items = pd.read_excel("/content/Nautica-Home-Project/Nautica Home - Items Catalog Clean.xlsx")
df_items.shape

(1401, 22)

This Dataframe has now 1401 rows over 22 columns, so we deleted 28 columns and 178 rows.


# Second Table - 'Orders'

## 1. Loading the Table

In [23]:
df_orders = pd.read_excel("/content/Nautica-Home-Project/Orders.xlsx")

In [24]:
print(f"This Dataframe has {df_orders.shape[0]} rows over {df_orders.shape[1]} columns")

This Dataframe has 7412 rows over 6 columns


In [25]:
df_orders.head()

Order_Date  Customer_ID           City    Order_ID Catalog_Number  Quantity
0 2022-03-28       100073  תל אביב - יפו  SO22000056            100         1
1 2022-03-28       100073  תל אביב - יפו  SO22000056   NHT3004-9-16         1
2 2022-03-28       100073  תל אביב - יפו  SO22000056  NHT3007-15-11         1
3 2022-04-03       100111         הרצליה  SO22000099            100         1
4 2022-04-03       100111         הרצליה  SO22000099   NHB5080-16-3         1

## 2. Cleanup the Table

### 2.1 Empty columns and rows

In [26]:
df_orders.isnull().sum()

Order_Date        0
Customer_ID       0
City              3
Order_ID          0
Catalog_Number    0
Quantity          0
dtype: int64

Note : This dataset doesn't contain **completely** NaN columns or rows.☝

### 2.2 Columns with single value

In [27]:
single_value_col = df_orders.columns[df_orders.nunique() <=1]
single_value_col

Index([], dtype='object')

Note : The dataset doesn't contain columns with single value

### 2.3 Duplicate Rows

In [28]:
duplicate_rows = df_orders[df_orders.duplicated()].shape[0]
print(f"This Dataframe has {duplicate_rows} duplicate rows ")

This Dataframe has 0 duplicate rows 


### 2.4 Drop unnecessary columns and rows

In [29]:
filter = df_orders.isnull().sum().sort_values(ascending=False)
filter[ filter > 0 ]

City    3
dtype: int64

In [30]:
filter_1 = (df_orders.isnull().sum()/df_orders.shape[0]).sort_values(ascending=False)
filter_1[ filter_1 > 0  ] # percentage of NaNs value

City    0.000405
dtype: float64

In order to continue the process of cleaning the data, we met with the CTO and presented to him:


*   Columns with missing values.
*   Columns with values thaht do not make sense.

After this meeting we decided to :
*   Remove the rows where `Catalog_Number` is equal to `100`.
*   Remove the rows where `Catalog_Number` is equal to `200`.

So we get the following data  👇

In [31]:
remove = df_orders['Catalog_Number'].isin([100,200])
df_orders = df_orders[~remove]
df_orders

Order_Date  Customer_ID           City    Order_ID Catalog_Number  \
1    2022-03-28       100073  תל אביב - יפו  SO22000056   NHT3004-9-16   
2    2022-03-28       100073  תל אביב - יפו  SO22000056  NHT3007-15-11   
4    2022-04-03       100111         הרצליה  SO22000099   NHB5080-16-3   
5    2022-04-03       100111         הרצליה  SO22000099  NHB5195-16-25   
6    2022-04-03       100111         הרצליה  SO22000099   NHB5340-16-3   
...         ...          ...            ...         ...            ...   
7404 2023-01-24       102184         הרצליה  SO23000265  NBRK403-10-40   
7406 2023-01-24       102185          רעננה  SO23000266     NLW-7070M3   
7408 2023-01-24       102186      פתח תקווה  SO23000267   NLW-71002XS6   
7409 2023-01-24       102186      פתח תקווה  SO23000267    NLW-71003S6   
7411 2023-01-24       102187      עין הבשור  SO23000268  NHB5300-16-11   

      Quantity  
1            1  
2            1  
4            1  
5            1  
6            1  
...        ...  
7404         2  
7406         1  
7408         1  
7409         1  
7411         1  

[5265 rows x 6 columns]

In [32]:
df_orders.shape

(5265, 6)

This Dataframe has now 5265 rows over 6 columns, so we deleted 0 columns and 2147 rows.

In [33]:
df_orders.isnull().sum()

Order_Date        0
Customer_ID       0
City              2
Order_ID          0
Catalog_Number    0
Quantity          0
dtype: int64